experiments/experiment.py

```diff
@@ -145,42 +147,24 @@ def run(self):
             for track_stat in self.stat_trackers:
                 track_stat(state, {"key": "T_END"})
 
-            # TODO: Stream back stats
-            # For now: write to file
-            self.write(self.generate_name(), state)
-
-    def format_data(self, data):
-        # state = {
-        #     "cat": cat,
-        #     "dai": dai,
-        #     "dai_join": dai_join,
-        #     "flapper": flapper,
-        #     "flippers": flippers,
-        #     "flopper": flopper,
-        #     "gem_joins": gem_joins,
-        #     "ilks": ilks,
-        #     "keepers": keepers,
-        #     "spotter": spotter,
-        #     "stats": {},
-        #     "vat": vat,
-        #     "vow": vow,
-        # }
+            self.write(self.generate_name(), state, _t)
+
+    def format_data(self, state, full_state=True):
+        data = state if full_state else state["stats"]
+        data = copy.deepcopy(data)
         for key, value in data.items():
-            # serialize stats
-            if key == "stats":
-                pass
-            # default behavior: get id of contracts
-            else:
-                data[key] = value.ADDRESS
+            if isinstance(value, (Ray, Rad, Wad)):
+                data[key] = float(data[value])
+            elif isinstance(value, dict):
```

Comment:
let's add a final case for iterables - as it stands, if we had an array `[Wad(1), Wad(2), Wad(3)]`, it would fail both of the existing cases and get untouched in the formatting

Reply:
other than that this looks good to merge